lib/IRGen/GenDecl.cpp

```diff
@@ -5850,3 +5852,24 @@ IRGenModule::getOrCreateHelperFunction(StringRef fnName, llvm::Type *resultTy,
 
   return fn;
 }
+
+void IRGenModule::setColocateMetadataSection(llvm::Function *f) {
+  switch (TargetInfo.OutputObjectFormat) {
+  case llvm::Triple::MachO:
+    f->setSection("__TEXT, __swift_colocat, regular");
+    break;
+  case llvm::Triple::DXContainer:
+  case llvm::Triple::GOFF:
+  case llvm::Triple::SPIRV:
+  case llvm::Triple::UnknownObjectFormat:
+    break;
+  case llvm::Triple::Wasm:
+  case llvm::Triple::ELF:
+    f->setSection("swift_colocate_functions");
+    break;
+  case llvm::Triple::XCOFF:
+  case llvm::Triple::COFF:
+    f->setSection(".sw5colo$B");
```

Comment:
Do we intend to need to create a linker table for this data?  The grouping (`$B`) is for cases where we need to order the data in between start/stop markers to allow the table computation.  If so, we should probably have an associated set of changes to `SwiftRT-ELF.cpp` and `SwiftRT-COFF.cpp`.

Reply:
No. I just copied the string from further up the file.

Reply:
In that case, this should probably be named `.sw5colo`.

Reply:
👍 will do